shoop/simple_cms/plugins.py

```diff
@@ -77,14 +77,14 @@ def get_context_data(self, context):
         selected_pages = self.config.get("pages", [])
         show_all_pages = self.config.get("show_all_pages", True)
         hide_expired = self.config.get("hide_expired", False)
+
         pages_qs = Page.objects.filter(visible_in_menu=True)
-        if not show_all_pages:
-            if selected_pages:
-                pages_qs = pages_qs.filter(id__in=selected_pages),
-            else:
-                pages_qs = pages_qs.none()
         if hide_expired:
             pages_qs = pages_qs.visible()
+
+        if not show_all_pages:
+            pages_qs = pages_qs.filter(id__in=selected_pages)
+
```

Comment:
Does this give you any problems when show_all_pages is turned off and you have no pages selected? I had a weird problem where id__in=[] was returning an queryset with an empty list or something strange, and the ``[]`` was rendering on the page.

If that hasn't been a problem for you, though, it should be fine.